base/src/main/java/io/spine/code/proto/EntityOption.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.code.proto;
+
+import com.google.protobuf.Descriptors.Descriptor;
+import io.spine.option.OptionsProto;
+
+import java.util.Optional;
+
+/**
+ * An option that, for a message representing a state of the entity, defines its kind (of which
+ * there are four: Aggregate, Projection, Process Manager and Entity) and its visibility of the
+ * entity to queries.
+ */
+public class EntityOption extends MessageOption<io.spine.option.EntityOption> {
```

Comment:
Not sure about the naming here - naming it `Entity` is misleading, but the clash with `io.spine.option.EntityOption` (which is just a composite message representation with `Kind` and `Visibility`) hurts the way user code looks.

Reply:
How about naming it `EntityStateOption`? The option describes the entity _state_ message, not the whole entity.